x-pack/filebeat/input/awss3/acks.go

```diff
@@ -0,0 +1,86 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package awss3
+
+import (
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common/acker"
+	"github.com/elastic/beats/v7/libbeat/common/fifo"
+)
+
+type awsACKHandler struct {
+	pending    fifo.FIFO[pendingACK]
+	ackedCount int
+
+	pendingChan chan pendingACK
+	ackChan     chan int
+}
+
+type pendingACK struct {
+	eventCount  int
+	ackCallback func()
+}
+
+func newAWSACKHandler() *awsACKHandler {
+	handler := &awsACKHandler{
+		pendingChan: make(chan pendingACK, 10),
+		ackChan:     make(chan int, 10),
```

Comment:
At a minimum I think we need a comment on why `10`.  But I've got a couple of questions.  Do these always need to be in sync?  Would we ever want to change these?